src/PollySandbox/ResilienceStrategyFactory.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Martin Costello, 2023. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+
+using Polly;
+using Polly.CircuitBreaker;
+using Polly.Fallback;
+using Polly.Timeout;
+using Refit;
+
+namespace PollySandbox;
+
+public partial class ResilienceStrategyFactory
+{
+    private readonly ResilienceStrategyStore _store;
+    private readonly IMetricsPublisher _metricsPublisher;
+    private readonly ILogger _logger;
+
+    public ResilienceStrategyFactory(
+        ResilienceStrategyStore store,
+        IMetricsPublisher metricsPublisher,
+        ILogger<ResilienceStrategyFactory> logger)
+    {
+        _store = store;
+        _metricsPublisher = metricsPublisher;
+        _logger = logger;
+    }
+
+    public ResilienceStrategy GetStrategy(string token, ApiEndpointOption endpoint, string resource)
+        => _store.GetStrategy(token, endpoint, resource);
+
+    public ResilienceStrategy<TResult> GetStrategy<TResult>(
```

Comment:
This method is executed on hot path and always creates new resilience strategy which is not very efficient. 

Is it possible to cache the strategy?

Reply:
It _should_ be only doing this for dynamic fallbacks layered on top of the rest of the strategies coming from the store.

Reply:
You could attach `ClientExecuteOptions` into `ResilienceContext` and just read it from cached strategies.

`ResilienceStrategyRegistry` allows adding new manually created strategies at runtime.